internal/local/environment.go

```diff
@@ -0,0 +1,56 @@
+package local
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/10gen/realm-cli/internal/cli/user"
+)
+
+// Environment contains the name, full filepath, and api keys of a profile
+type Environment struct {
+	Name        string
+	Filepath    string
+	Credentials user.Credentials
+}
+
+// LoadProfileEnvs returns a list of each profile's environment containing name, filepath, and api keys
+func LoadProfileEnvs() ([]Environment, error) {
+	dir, dirErr := user.HomeDir()
+	if dirErr != nil {
+		return nil, fmt.Errorf("failed to get CLI profiles: %w", dirErr)
+	}
+
+	dirEntryList, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get CLI profiles: %w", dirErr)
+	}
+
+	environments := make([]Environment, 0, len(dirEntryList))
+	for _, v := range dirEntryList {
+		if !strings.Contains(v.Name(), user.ProfileType) {
+			continue
+		}
+
+		name := strings.TrimSuffix(v.Name(), filepath.Ext(v.Name()))
+
+		profile, err := user.NewProfile(name)
+		if err != nil {
+			return nil, fmt.Errorf("failed to load profile: %s", profile.Name)
+		}
+		if err := profile.Load(); err != nil {
```

Comment:
there might be unforeseen ramifications using `viper` to `Load` each profile individually (there might be unforeseen ramifications to _not_ using `viper` to `Load` each profile 😬 )

due to the command we're supporting here, I'd rather us take the simple/naive approach first before we get too clever with it

as such, we should just define `yaml` tags on w/e `Credentials` struct we end up using and get that data via `yaml.Unmarshal`

i'll leave it to you if we end up adding `yaml` tags to `user.Credentials` type or define an entirely new type or set of fields here in or alongside `Environment`